### PR TITLE
gh-101372: Fix unicodedata.is_normalized to properly handle the UCD 3…

### PR DESCRIPTION
**Pull request by corona10**
….2.0

* Issue: gh-101372

### CODE REVIEW

File: Lib/test/test_unicodedata.py [hunk 1]
```diff
@@ -216,6 +216,12 @@ def test_issue29456(self):
         self.assertEqual(self.db.normalize('NFC', u11a7_str_a), u11a7_str_b)
         self.assertEqual(self.db.normalize('NFC', u11c3_str_a), u11c3_str_b)
 
+    def test_is_normalized_unicode_3_2_0(self):
```
Comment (serhiy-storchaka) [inline Lib/test/test_unicodedata.py]: How slow is this test? If it takes a significant amount of time (in comparison with other tests), perhaps we only need to test few example cases.

I would test cases which return different results for 3.2.0 and the current database. Also, is it worth to include tests for multicharacter strings? Maybe take some examples from other tests for `normalize()` and `is_normalized()`.

Comment (corona10) [reply Lib/test/test_unicodedata.py]: > I would test cases which return different results for 3.2.0 and the current database. Also, is it worth to include tests > for multicharacter strings? Maybe take some examples from other tests for normalize() and is_normalized().

Okay, I will try it!

**Comment by corona10** [comment]
@serhiy-storchaka 

All ranges of characters are candidates for testing.
So I decide to choose a sampling approach rather than choose specific cases.


## Test script
````python
import unicodedata

with open('foo.out', 'w') as f:
    for x in range(0x110000):
        for form in ('NFC', 'NFD', 'NFKC', 'NFKD'):
            norm = unicodedata.ucd_3_2_0.normalize(form, chr(x))
            if not unicodedata.ucd_3_2_0.is_normalized(form, norm):
                f.write(f'{str(x)},{form}\n')
````

## AS-IS
````
(.oss) ➜  cpython git:(main) ✗ ./python.exe gh-101372.py
(.oss) ➜  cpython git:(main) ✗ wc -l foo.out
 4456448 foo.out
````

## TO-BE
````

(.oss) ➜  cpython git:(gh-101372) ✗ ./python.exe gh-101372.py
(.oss) ➜  cpython git:(gh-101372) ✗ wc -l foo.out
       0 foo.out
````

**Comment by corona10** [comment]
@serhiy-storchaka I will merge this PR by next week, please let me know if there need some changes

**Comment by serhiy-storchaka** [comment]
I am not happy with provided tests.

Testing all range of Unicode characters is slow (few seconds on my computer), it should be decorated with `@requires_resource('cpu')` if performed. Testing only small random sample can miss errors, and the test result will be hardly reproducible. It works for this issue, because `is_normalized()` was broken for most of characters, but it could not work for other types of bugs.

The test for multicharacter string is not what I meant. It should not only test all normalized sequences, but also non-normalized sequences. For example, `'\ufb2c'` is normalized to `'\u05e9\u05bc\u05c1'`. Therefore, `'\ufb2c'` should be not normalized, and `'\u05e9\u05bc\u05c1'` should be normalized. But `'\u05e9\u05c1\u05bc'`, created by swapping the last two characters, is normalized to the same sequence `'\u05e9\u05bc\u05c1'`, therefore it should be not normalized, besides it looks exactly the same as the original character. I think we need such kind of tests.

I tried to write more interesting tests for `is_normalized()`, and have found that the UCD 3.2.0 is mostly not tested. Also, there are not many tests for differences between UCD 3.2.0 and the current version. I am writing new tests.

I propose to merge your PR without tests. The bugfix itself is obvious, and the tests I will add later.

**Comment by corona10** [comment]
> I propose to merge your PR without tests. The bugfix itself is obvious, and the tests I will add later.

Okay got it, Please let me know once you submit test patches. I may learn a lot from the patch.

**Pull request merged**